core/chains/evm/config/toml/defaults.go

```diff
@@ -65,6 +74,64 @@ func init() {
 	slices.SortFunc(DefaultIDs, func(a, b *big.Big) int {
 		return a.Cmp(b)
 	})
+
+	// read the custom defaults overrides
+	dir := os.Getenv(CUSTOM_DEFAULTS_ENV_KEY)
+	if dir == "" {
+		// short-circuit; no default overrides provided
+		return
+	}
+
+	// Read directory contents
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		fmt.Println("Error reading directory:", err)
+		return
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() {
+			// Skip directories
+			continue
+		}
+
+		path := dir + "/" + entry.Name()
+		file, err := os.Open(path)
+		if err != nil {
+			fmt.Println("Error opening file:", err)
+			continue
+		}
+
+		// Read file contents
+		b, err := io.ReadAll(file)
+		if err != nil {
+			fmt.Println("Error reading file:", err)
+			file.Close()
+			continue
+		}
+		file.Close()
+
+		var config = struct {
+			ChainID *big.Big
+			Chain
+		}{}
+
+		if err := cconfig.DecodeTOML(bytes.NewReader(b), &config); err != nil {
+			log.Fatalf("failed to decode %q: %v", path, err)
+		}
+
+		if config.ChainID == nil {
+			log.Fatalf("missing ChainID: %s", path)
+		}
+
+		id := config.ChainID.String()
+		// these custom defaults are only meant to be overrides on existing chains;
+		// if the chain does not exist already in defaults error out
```

Comment:
It would probably be OK to relax this

Reply:
I'd rather start with tighter restrictions to start and then loosen if we need

Reply:
This will require CCIP to issue a new release in order to configure an unknown chain ID. We have never had that restriction in place with normal config.toml files, so it seems inconsistent and surprising.

Reply:
In fact, we don't even necessarily have overridden defaults for all supports chains, so I think it is not actually viable without other extra changes to track an allowed set of chain IDs.

Reply:
>In fact, we don't even necessarily have overridden defaults for all supports chains, so I think it is not actually viable without other extra changes to track an allowed set of chain IDs.

👍 I'll loosen the restriction